EyeOnBuzz/Infrastructure/DataSources/GenresDataSource.swift

```diff
@@ -0,0 +1,47 @@
+//
+//  GenresDataSource.swift
+//  EyeOnBuzz
+//
+//  Created by Weverton Couto Timoteo on 1/28/18.
+//  Copyright © 2018 Weverton Couto Timoteo. All rights reserved.
+//
+
+import Foundation
+
+class GenresDataSource {
+    
+    let session = TMDBSession.init()
+    let resultsNodeName = "genres"
+    let dataSourceTarget: DataSourceTarget
+    
+    var list = [Int: String]()
+    
+    init(dataSourceTarget: DataSourceTarget) {
+        self.dataSourceTarget = dataSourceTarget
+    }
+    
+    func fetch() {
+        self.session.get(url: Router.TheMovieDatabaseAPI.genres, parameters: nil, success: { statusCode, headers, response in
+            DispatchQueue.main.async {
+                if let fullResponse = response as? Dictionary<String, Any> {
```

Comment:
You are processing the response and parsing the objects on the main thread here. Do you think this might cause you any problems?

Reply:
Yes, when parsing response in main thread, it will be blocked to other process such as UI update. I would improve it to process response in background and just trigger notification center that list was updated.